artifactory/federation.go

```diff
@@ -53,34 +78,31 @@ func (c *Client) FetchMirrorLags() (MirrorLags, error) {
 	return mirrorLags, nil
 }
 
-// UnavailableMirror represents single element of API respond from federation/status/unavailableMirrors endpoint
-type UnavailableMirror struct {
-	LocalRepoKey  string `json:"localRepoKey"`
-	RemoteUrl     string `json:"remoteUrl"`
-	RemoteRepoKey string `json:"remoteRepoKey"`
-	Status        string `json:"status"`
-}
-
-type UnavailableMirrors struct {
-	UnavailableMirrors []UnavailableMirror
-	NodeId             string
-}
-
 // FetchUnavailableMirrors makes the API call to federation/status/unavailableMirrors endpoint and returns []UnavailableMirror
 func (c *Client) FetchUnavailableMirrors() (UnavailableMirrors, error) {
 	var unavailableMirrors UnavailableMirrors
 	c.logger.Debug("Fetching unavailable mirrors")
-	resp, err := c.FetchHTTP(federationUnavailableMirrorsEndpoint)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second) // Set a timeout duration
+	defer cancel()
+
+	resp, err := c.FetchHTTPWithContext(ctx, federationUnavailableMirrorsEndpoint) // Assume FetchHTTPWithContext handles context
```

Comment:
@chewrocca, if you have provided `FetchHTTPWithContext` please consider removing comment in this line.